src/Stratis.Bitcoin.Features.MemoryPool.Tests/MempoolValidatorTest.cs

```diff
@@ -484,10 +795,29 @@ public void AcceptToMemoryPool_TxAlreadyHaveCoins_ReturnsFalse()
             // TODO: Execute this case CheckMempoolCoinView context.View.HaveCoins(context.TransactionHash)
         }
 
-        [Fact(Skip = "Not implemented yet.")]
-        public void AcceptToMemoryPool_TxMissingInputs_ReturnsFalse()
+        [Fact]
+        public async void AcceptToMemoryPool_TxMissingInputs_ReturnsFalseAsync()
         {
-            // TODO: Execute this case CheckMempoolCoinView !context.View.HaveCoins(txin.PrevOut.Hash)
+            string dataDir = GetTestDirectoryPath(this);
+
+            var minerSecret = new BitcoinSecret(new Key(), KnownNetworks.RegTest);
+            ITestChainContext context = await TestChainFactory.CreateAsync(KnownNetworks.RegTest, minerSecret.PubKey.Hash.ScriptPubKey, dataDir);
+            IMempoolValidator validator = context.MempoolValidator;
+            Assert.NotNull(validator);
+
+            var destSecret = new BitcoinSecret(new Key(), KnownNetworks.RegTest);
+            var tx = new Transaction();
+
+            // Nonexistent input.
+            tx.AddInput(new TxIn(new OutPoint(new uint256(57), 0), PayToPubkeyHashTemplate.Instance.GenerateScriptPubKey(minerSecret.PubKey)));
+            tx.AddOutput(new TxOut(new Money(Money.CENT * 11), destSecret.PubKeyHash));
+            tx.Sign(KnownNetworks.RegTest, minerSecret, false);
+
+            var state = new MempoolValidationState(false);
+
+            // CheckMempoolCoinView !context.View.HaveCoins(txin.PrevOut.Hash)
+            bool isSuccess = await validator.AcceptToMemoryPool(state, tx);
+            Assert.False(isSuccess, "Transaction with invalid input should not have been accepted.");
```

Comment:
Please specify the error here as well, and also all other tests, this is fairly important.

Reply:
Yeah good spot, I think I addressed this in the changes that will form the second PR, but I will copy the critical fixes in here.